### DISCUSSION TITLE
LazyMemorySegmentPool use BlockingQueue instead of using list with lock

### DISCUSSION THREAD

**Discussion by swuferhong** [apache/fluss#90] [General]
Currently, in `LazyMemorySegmentPool`, we use a List with a lock to manage cachePages. This approach requires the use of locks and manual writing await logic, which is not as efficient as using a `BlockingQueue`.  Moreover, replacing it with a `BlockingQueue` would not introduce any potential correctness issues.

**Comment by 0rrange** [comment]
I would like to do it.pls assign to me

**Reply by swuferhong** [reply]
Hi, @0rrange you can submit a PR,  after submitting, @ me to help with the review.

**Reply by 0rrange** [reply]
ok

**Comment by wuchong** [comment]
convert to issue #136
